communities/src/lib.rs

```diff
@@ -92,15 +113,31 @@ decl_module! {
                 validate_nominal_income(&i).map_err(|_| <Error<T>>::InvalidNominalIncome)?;
             }
 
-            let cid = CommunityIdentifier::from(blake2_256(&(loc.clone(), bootstrappers.clone()).encode()));
+            let cid = CommunityIdentifier::from(blake2_256(&(location.clone(), bootstrappers.clone()).encode()));
             let cids = Self::community_identifiers();
             ensure!(!cids.contains(&cid), "community already registered");
-            Self::validate_locations(&loc, &cids)?; // <- O(n^2) !!!
 
+            Self::validate_location(&location)?;
             // All checks done, now mutate state
+            let geo_hash = encode(location.lat, location.lon, GEO_HASH_LENGTH).map_err(|_| <Error<T>>::InvalidLocationForGeohash)?;
+            let mut locations: Vec<Location> = Vec::new();
+
+            // insert cid into cids_by_geohash map
+            let mut cids_in_bucket = Self::cids_by_geohash(geo_hash.to_pallet_string());
+            match cids_in_bucket.binary_search(&cid) {
```

Comment:
is this better than `.contains()`? more performant?

Reply:
I'd rather use `if` than `match`

Reply:
i used what substrate proposed [here](https://substrate.dev/recipes/vec-set.html)

Reply:
Although verbose, I reckon this is indeed more efficient.